src/GameLogic/PlayerActions/Items/ItemCraftAction.cs

```diff
@@ -55,6 +55,41 @@ await player.InvokeViewPlugInAsync<IShowMerchantStoreItemListPlugIn>(
             .ConfigureAwait(false);
     }
 
+    /// <summary>
+    /// Finds the relevant <see cref="ItemCrafting"/> by testing the mix items against every crafting's item requirements.
+    /// </summary>
+    /// <param name="player">The player.</param>
+    /// <returns>The relevant <see cref="ItemCrafting"/>.</returns>
+    public ItemCrafting? FindAppropriateCraftingByItems(Player player)
+    {
+        ItemCrafting? selectedCrafting = null;
+
+        if (player.OpenedNpc?.Definition is { } npc)
+        {
+            foreach (var itemCrafting in npc.ItemCraftings)
+            {
+                if (!this._craftingHandlerCache.TryGetValue(itemCrafting, out var craftingHandler))
+                {
+                    craftingHandler = this.CreateCraftingHandler(itemCrafting);
+                    this._craftingHandlerCache.Add(itemCrafting, craftingHandler);
+                }
+
+                if (craftingHandler.TryGetRequiredItems(player, out _, out _) is null)
+                {
+                    selectedCrafting = itemCrafting;
+                    if (selectedCrafting.Number != 1)
+                    {
+                        // If any but the chaos weapon crafting, the search is finished.
+                        // Otherwise, it still could be a wing crafting, which has an extra item requirement (chaos weapon).
+                        break;
```

Comment:
I think then the code must make sure that the wings are checked after the chaos weapons. It could make sense to add a `OrderBy(c => c.Number)` to the `npc.ItemCraftings` before processing them.